test_dev.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+# Catch any non-flagged use
+if [ $# -eq 0 ]; then
+    echo "Invalid: $(basename "$0") has required flags, use: -h for help"
+    exit 1
+fi
+
+# e: Stops execution in the instance of a command or pipeline error
+# u: Treat unset variables as an error and exit immediately
+set -eu
+
+# Default to FAILURE
+STATUS="FAILURE"
+
+function outcome_and_exit {
+  if [ $STATUS == "SUCCESS" ]
+  then
+    echo -e "\e[32m$STATUS\e[0m"
+  else
+    echo -e "\e[31m$STATUS\e[0m"
+  fi
+}
+
+while getopts ":ha" opt; do # Parse options to the `test_cmd` command
+  case ${opt} in
+    h)
+      echo
+      echo "Usage:"
+      echo "    -h             Displays this help message"
+      echo "    -a             Run all unit and integration tests"
+      echo
+      echo "Commands:"
+      echo "    unit           Unit tests subcommand, run unit -h for more information"
+      echo "    integration    Integration tests subcommand, run integration -h for more information"
+      exit 0
+      ;;
+    a) # -a (all)
+      trap "outcome_and_exit" EXIT
+      docker-compose run --use-aliases boulder ./test.sh
+      ;;
+    *) # catch invalid options
+      echo "Invalid Option: $OPTARG use: -h for help" 1>&2
+      exit 1
+      ;;
+  esac
+done
+
+if [ $# -gt 1 ] # Check for subcommands
+then
+  shift $((OPTIND -1))
+  subcommand="${1}"; shift
+
+  case "$subcommand" in
+  unit) # Parse options to the unit sub command
+    while getopts ":d:ha" opt; do
+      case ${opt} in
+        h) # -h (help)
+          echo
+          echo "Usage:"
+          echo "    -h                Displays this help message"
+          echo "    -a                Run all unit tests"
+          echo "    -d <DIRECTORY>    Run unit tests for a specific directory"
+          exit 0
+          ;;
+        a)  # -a (all)
+          trap "outcome_and_exit" EXIT
+          docker-compose run -e RUN="unit" --use-aliases boulder ./test.sh
```

Comment:
Consider skipping test.sh and just running what it runs, like you do in the integration tests: `go test -p 1 ./...`